### ISSUE TITLE
Please always support binary key files with a size of exactly 32 bytes

### ISSUE THREAD

**Issue by ky-ex** [keepassxreboot/keepassxc#7217]
Today, if kepassXC loads a key file of exactly 32 bytes, it allows it, but it will complain that it is a legacy key file format.
I think there is no good reason to complain about that file.

I suggest to change the code to no longer show a warning for key files that are exactly 32 bytes.

I looked at the code, and the mechanism that is considered "modern" is apparently to support using any kind of file, and create a SHA256 hash, which also produces exactly 32 bytes.

The only benefit of using SHA256 is to derive the arbitrary input file to something that is 32 bytes. I don't see any benefit of applying this strategy when the file has exactly 32 bytes.

However, there is a benefit of keeping the exception, and use the key file directly if it's size is exactly 32 bytes.

The benefit is, expert users can avoid the manual conversion process of saving the database in a secure place, removing the key file, and generating a new keyfile.

Expert users can manually convert the key file of the old format to the 32 bytes binary file.

The benefit is, the same database file can be opened with both old application versions and the old style key file format, and also with new keepassXC versions with a converted key file.

In other words, the database file doesn't need conversion and can remain the same across versions, and only the key file needs conversion.

**Comment by droidmonkey** [comment]
We specifically discourage this old key file type because it is completely arbitrary. The only differentiator between the "any file that is hashed" and the 32 byte key file is the fact that it is exactly 32 bytes. We don't want you using this and we will probably remove support for it in the future. 

Further, you will lose compatibility with older versions of keepassx due to up conversion to kdbx 4 if you use any newer features within keepassxc. 

Also, the warning has a checkbox that let's you suppress the warning. So just check the checkbox.
